Microsoft.Azure.Cosmos/src/Diagnostics/CosmosDiagnostics.cs

```diff
@@ -23,6 +23,26 @@ public virtual TimeSpan GetClientElapsedTime()
             throw new NotImplementedException($"CosmosDiagnostics.GetElapsedTime");
         }
 
+        /// <summary>
+        /// This represent the start time of the request.
+        /// </summary>
+        /// <returns>This returns the start time of the request.</returns>
+        public virtual DateTime GetStartTimeUtc()
+        {            
+            // Default implementation avoids breaking change for users upgrading.
+            throw new NotImplementedException($"CosmosDiagnostics.GetStartTimeUtc");
+        }
+
+        /// <summary>
+        ///  This represent the count of failed requests.
+        /// </summary>
+        /// <returns>The integer field returns the count of failed requests.</returns>
+        public virtual int FailedRequestCount()
```

Comment:
Remove default implementation to follow like GetContactedRegions. It helps avoid run time exceptions.

```suggestion
        public abstract int FailedRequestCount();
```